config.py

```diff
@@ -1299,3 +1308,37 @@ def initialize_logging(logger_name):
     # re-get the log after logging is initialized
     global log
     log = logging.getLogger(__name__)
+
+def setup_temp_dir(agentConfig, proc_name=None):
```

Comment:
Yep, keeping the branching inside is fine, I'm just not comfortable with passing the whole agentConfig (I find it harder to understand at a glance what setup_temp_dir does).
What do you think of `def setup_temp_dir(use_dd_temp_dir, custom_temp_dir, proc_name=None):` and pass it as `setup_temp_dir(agentConfig['use_dd_temp_dir'], agentConfig['custom_temp_dir'], proc_name='forwarder')` ?